### PR TITLE
sap_hypervisor_node_preconfigure: Fixed role name and path for include tasks

### PR DESCRIPTION
**Pull request by newkit**
When testing the ansibe-galaxy community version, a couple of issues occurred that are fixed with this PR.

1. Role name is community.sap_install.sap_hypervisor_node_preconfigure (adjusted in example playbook)
2. When doing include_tasks, the path has to be relative to the role top tasks directory.
3. Reads the KUBECONFIG environment variable and sets the environment for the playbook accordingly.

### CODE REVIEW

File: playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml [hunk 1]
```diff
@@ -4,8 +4,9 @@
...
-  tasks:
-    - name: Include Role
-      ansible.builtin.include_role:
-        name: sap_hypervisor_node_preconfigure
+    sap_hypervisor_node_kubeconfig: "{{ lookup( 'ansible.builtin.env', 'KUBECONFIG') }}"
+  environment:
+    KUBECONFIG: "{{ sap_hypervisor_node_kubeconfig }}"
+    K8S_AUTH_KUBECONFIG: "{{ sap_hypervisor_node_kubeconfig }}"
+  roles:
+    - { role: community.sap_install.sap_hypervisor_node_preconfigure }
```
Comment (kksat) [inline playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml:12]: this is not correct syntax, should be 
```ansible 
  roles:
   - community.sap_install.sap_hypervisor_node_preconfigure
```

File: playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml [hunk 2]
```diff
@@ -4,8 +4,9 @@
...
     sap_hypervisor_node_platform: redhat_ocp_virt
-
-  tasks:
-    - name: Include Role
-      ansible.builtin.include_role:
-        name: sap_hypervisor_node_preconfigure
+    sap_hypervisor_node_kubeconfig: "{{ lookup( 'ansible.builtin.env', 'KUBECONFIG') }}"
+  environment:
+    KUBECONFIG: "{{ sap_hypervisor_node_kubeconfig }}"
+    K8S_AUTH_KUBECONFIG: "{{ sap_hypervisor_node_kubeconfig }}"
```
Comment (kksat) [inline playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml:10]: I am no k8s expert, just double checking K8S_AUTH_KUBECONFIG should have the same value as KUBECONFIG?

File: playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml [hunk 3]
```diff
@@ -4,8 +4,9 @@
   serial: 1
   vars:
     sap_hypervisor_node_platform: redhat_ocp_virt
-
-  tasks:
-    - name: Include Role
-      ansible.builtin.include_role:
-        name: sap_hypervisor_node_preconfigure
+    sap_hypervisor_node_kubeconfig: "{{ lookup( 'ansible.builtin.env', 'KUBECONFIG') }}"
```
Comment (kksat) [inline playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml:7]: Could you please elaborate why double assignment needed for KUBECONFIG? Ansible process will start environment variables set for the parent process. So I do not see why this double assignment is required.

File: playbooks/vars/sample-variables-sap-hypervisor-node-preconfigure-rh_ocp_virt.yml [hunk 4]
```diff
@@ -1,3 +1,8 @@
+sap_hypervisor_node_preconfigure_install_operators: true
```
Comment (kksat) [inline playbooks/vars/sample-variables-sap-hypervisor-node-preconfigure-rh_ocp_virt.yml:1]: I see these variables are not documented, suggestions is to set defaults and document them

File: roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/main.yml [hunk 5]
```diff
@@ -61,22 +61,22 @@
   when: __sap_hypervisor_node_preconfigure_register_worker_memory_gib | int >= 512
 
 - name: Include prepare
-  ansible.builtin.include_tasks: prepare.yml
+  ansible.builtin.include_tasks: "platform/{{ sap_hypervisor_node_platform }}/prepare.yml"
```
Comment (kksat) [inline roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/main.yml:64]: I see that folders respective to possible platforms do not have all the task files.
I think this is just an oversight. 
Maybe worth adding content to that folders to this PR.

**Review by kksat — COMMENTED** [review]
just some comments

File: playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml (hunk 1 above)
Comment (berndfinger) [reply playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml:12]: @kksat Please provide evidence to support your statement. The latest [Using Ansible playbooks -> Working with playbooks -> Roles](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html) document contains the  syntax you rejected in [this section](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html#id12). To me, it appears that this syntax is not deprecated or forbidden and we can safely leave it in place.

Comment (kksat) [reply playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml:12]: my mistake, we of course can leave this as it is.

File: playbooks/vars/sample-variables-sap-hypervisor-node-preconfigure-rh_ocp_virt.yml (hunk 4 above)
Comment (newkit) [reply playbooks/vars/sample-variables-sap-hypervisor-node-preconfigure-rh_ocp_virt.yml:1]: It is defined here:

`sap_hypervisor_node_preconfvars/platform_defaults_redhat_ocp_virtp_virt.yml

```
# Should the operators be installed
sap_hypervisor_node_preconfigure_install_operators: True
```

File: roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/main.yml (hunk 5 above)
Comment (newkit) [reply roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/main.yml:64]: The switch which platform to load is one level up in here `roles/sap_hypervisor_node_preconfigure/tasks/main.yml`. So I see no reason why all files would have to be available for other platforms there.

File: playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml (hunk 2 above)
Comment (newkit) [reply playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml:10]: Some modules / tasks use KUBECONFIG, some K8S_AUTH_KUBECONFIG. Thats why they _are_ set identically. The value is read from the environment from the calling environment and has to be set specifically for ansible.

File: playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml (hunk 3 above)
Comment (newkit) [reply playbooks/sample-sap-hypervisor-redhat_ocp_virt-preconfigure.yml:7]: Actually same as above, some modules / tasks use KUBECONFIG, some K8S_AUTH_KUBECONFIG. Thats why they _are_ set identically. The value is read from the environment from the calling environment and has to be set specifically for ansible.

**Review by berndfinger — APPROVED** [review]
LGTM

**Pull request merged**